qiskit_machine_learning/neural_networks/sampler_qnn.py

```diff
@@ -0,0 +1,402 @@
+        qc.compose(ansatz, inplace=True)
+
+
+        def parity(x):
+            return "{:b}".format(x).count("1") % 2
+
+
+        qnn = SamplerQNN(
+            circuit=qc,
+            input_params=feature_map.parameters,
+            weight_params=ansatz.parameters,
+            interpret=parity,
+            output_shape=2
+        )
+
+        qnn.forward(input_data=[1, 2], weights=[1, 2, 3, 4])
+
+    The following attributes can be set via the constructor but can also be read and
+    updated once the SamplerQNN object has been constructed.
+
+    Attributes:
+
+        sampler (BaseSampler): The sampler primitive used to compute the neural network's results.
+        gradient (BaseSamplerGradient): An optional sampler gradient to be used for the backward pass.
+    """
+
+    def __init__(
+        self,
+        *,
+        circuit: QuantumCircuit,
+        sampler: BaseSampler | None = None,
+        input_params: Sequence[Parameter] | None = None,
+        weight_params: Sequence[Parameter] | None = None,
+        sparse: bool = False,
+        interpret: Callable[[int], int | tuple[int, ...]] | None = None,
+        output_shape: int | tuple[int, ...] | None = None,
+        gradient: BaseSamplerGradient | None = None,
+        input_gradients: bool = False,
+    ):
+        """
+        Args:
+            sampler: The sampler primitive used to compute the neural network's results.
+                If ``None`` is given, a default instance of the reference sampler defined
+                by :class:`~qiskit.primitives.Sampler` will be used.
+            circuit: The parametrized quantum circuit that generates the samples of this network.
+            input_params: The parameters of the circuit corresponding to the input.
+            weight_params: The parameters of the circuit corresponding to the trainable weights.
+            sparse: Returns whether the output is sparse or not.
+            interpret: A callable that maps the measured integer to another unsigned integer or
+                tuple of unsigned integers. These are used as new indices for the (potentially
+                sparse) output array. If no interpret function is
+                passed, then an identity function will be used by this neural network.
+            output_shape: The output shape of the custom interpretation
+            gradient: An optional sampler gradient to be used for the backward pass.
+                If ``None`` is given, a default instance of
+                :class:`~qiskit.algorithms.gradients.ParamShiftSamplerGradient` will be used.
+            input_gradients: Determines whether to compute gradients with respect to input data.
+                 Note that this parameter is ``False`` by default, and must be explicitly set to
+                 ``True`` for a proper gradient computation when using ``TorchConnector``.
+        Raises:
+            QiskitMachineLearningError: Invalid parameter values.
+        """
```

Comment:
I will note that many of the types in the arg list are not clickable links here - just the purple colored ones. I am not sure whether its an aspect of the new syntax or not. If I look at CircuitQNN QuantumCircuit is link there whereas not here. I'll check with something newer from Nature, though one from 10 days ago has a similar issue in terms of some things are links others not.

Reply:
I don't know why as well. Perhaps due to the new syntax.